### PR TITLE
added delete project api endpoint

### PR DESCRIPTION
**Pull request by natalie-poulson**
This change addressed issue #103: Create API endpoint that would delete a project.

To use:
- POST http://127.0.0.1:5000/projects/<project_id>/delete

Contributors:
@natalie-poulson
@madelynnelson 
@littlemisslilycane

### CODE REVIEW

File: server/routes/github_oauth.js [hunk 1]
```diff
@@ -351,6 +351,23 @@ async function addNewProject(auth_token, position, experience_lvl, new_project)
...
+    .then(async (res) => {
+      return res;
+    })
+    .catch(err => {
+      throw err;
+    });
+};
+
+router.post("/projects/:id/delete", async (req,res) => {
+  const response = await deleteProject(req.params.id);
```
Comment (nurmerey) [inline server/routes/github_oauth.js:366]: Let's make sure to add check for authentication before deleting the project 😛. Please add following after L365:
 ```
 const auth_token = getToken(req.headers.authorization);
  if (auth_token === null) {
    return res.sendStatus(401);
  }
```

**Review by nurmerey — CHANGES_REQUESTED** [review]
@natalie-poulson @madelynnelson  @littlemisslilycane thank you for your contribution! This is looking good! Added a little ask in the comments!

File: server/routes/github_oauth.js (hunk 1 above)
Comment (natalie-poulson) [reply server/routes/github_oauth.js:370]: Good call! I added that in 👍

**Review by nurmerey — APPROVED** [review]
👍  LGTM! Thank you!

**Pull request merged**